### PR TITLE
Fix system library lookup for MacOS

### PR DESCRIPTION
**Pull request by LLITCHEV**
The compiler adds an underscore, but on MacOS system symbols have no one. The Stripe code did similar thing.
Thanks Mars!

### CODE REVIEW

File: pmlc/compiler/executable.cc [hunk 1]
```diff
@@ -164,7 +164,19 @@ static void *tryResolveSymbol(StringRef symbol) {
     if (auto ptr = resolveSymbol(symbol.drop_front()))
       return ptr;
   }
-  return llvm::sys::DynamicLibrary::SearchForAddressOfSymbol(symbol.str());
+
```
Comment (flaub) [inline pmlc/compiler/executable.cc:167]: Isn't this already handled in the few lines above?

**Comment by flaub** [comment]
Do you have `.mlir` test case where this currently fails?

**Comment by flaub** [comment]
Perhaps this was needed to find symbols for `sin`? If so, I'd like to explicitly register symbols in `pmlc/rt/runtime.cc` instead of trying to do an automatic lookup.

File: pmlc/compiler/executable.cc (hunk 1 above)
Comment (LLITCHEV) [reply pmlc/compiler/executable.cc:167]: The code just above handles resolving symbols from our library. There is no code to handle system library symbols like llvm::sys::DynamicLibrary::SearchForAddressOfSymbol("_cos"); for example. I merely added code to handle this case (as it was in the Stripe version).

**Comment by LLITCHEV** [comment]
Added an edsl test.

**Comment by LLITCHEV** [comment]
I tried registering a function, but it was always making a call - the cos function is intrinsic and the compiler replaces them in the LLVM phases with a CPU instruction. If we register a function, the function is always called (the compiler doesn't know how to replace it - that is what it did with all the attempts I tried to push it through) and thus creating a call overhead. Calling the intrinsic directly removes the need to have the call - better for perf. Maybe there is a way to register an intrinsic function which I just didn't see?

File: plaidml/edsl/tests/edsl_test.cc [hunk 2]
```diff
@@ -1090,6 +1090,30 @@ TEST_F(CppEdsl, PrngResultNotNegative) {
...
+  std::vector<float> input = {
+      5.0, 6.0, 7.0,  //
+      4.0, 5.0, 6.0,  //
+      7.0, 8.0, 9.0,  //
+  };
+
+  auto binder = exec::Binder(program);
+  auto executable = binder.compile();
+  binder.input(S).copy_from(input.data());
+  executable->run();
```
Comment (flaub) [inline plaidml/edsl/tests/edsl_test.cc:1114]: Use `checkProgram()`

File: pmlc/testing/BUILD [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# Copyright 2019 Intel Corporation.
```
Comment (flaub) [inline pmlc/testing/BUILD:1]: 2020

**Comment by flaub** [comment]
I think the long term resolution to all of this is to use the `llvm-project/rt` library directly. That's where these intrinsics are defined. But for now, we can use this approach.

File: plaidml/edsl/tests/edsl_test.cc (hunk 2 above)
Comment (LLITCHEV) [reply plaidml/edsl/tests/edsl_test.cc]: Using checkProgram.

File: pmlc/testing/BUILD (hunk 3 above)
Comment (LLITCHEV) [reply pmlc/testing/BUILD]: Fixed.

**Pull request merged**